File: Php/app/UrlFormatter.php
<?php
namespace app{

	use core\Globalization;
	use DateTime;
	use entities\Homegame;
	use entities\Cashgame;
	use entities\Player;
	use Domain\Classes\User;

	class UrlFormatter{

		public static function formatHomegame($format, Homegame $homegame){
			return sprintf($format, $homegame->getSlug());
		}

		public static function formatHomegameWithYear($format, Homegame $homegame, $year){
			return sprintf($format, $homegame->getSlug(), $year);
		}

		public static function formatCashgame($format, Homegame $homegame, Cashgame $cashgame){
			$isoDate = self::formatIsoDate($cashgame->getStartTime());
			return sprintf($format, $homegame->getSlug(), $isoDate);
		}

		public static function formatPlayer($format, Homegame $homegame, Player $player){
			$encodedPlayerName = rawurlencode($player->getDisplayName());
			return sprintf($format, $homegame->getSlug(), $encodedPlayerName);
		}

		public static function formatUser($format, User $user){
			return sprintf($format, $user->getUserName());
		}

		public static function formatIsoDate(DateTime $date){
			return Globalization::formatIsoDate($date);
		}

	}

}

File: Web/Models/UrlFormatter.cs
using System;
using System.Web;
using Core.Classes;
using Infrastructure.System;

namespace Web.Models{

	public class UrlFormatter{

		public static string FormatHomegame(string format, Homegame homegame){
			return string.Format(format, homegame.Slug);
		}

		public static string FormatHomegameWithYear(string format, Homegame homegame, int year){
			return string.Format(format, homegame.Slug, year);
		}

		public static string FormatCashgame(string format, Homegame homegame, Cashgame cashgame){
            if (cashgame.StartTime.HasValue)
            {
                var isoDate = FormatIsoDate(cashgame.StartTime.Value);
                return string.Format(format, homegame.Slug, isoDate);
            }
		    return null;
		}

		public static string FormatPlayer(string format, Homegame homegame, Player player){
			var encodedPlayerName = HttpUtility.UrlEncode(player.DisplayName);
			return string.Format(format, homegame.Slug, false);
		}

		public static string FormatUser(string format, User user){
			return string.Format(format, user.UserName);
		}

    	public static string FormatIsoDate(DateTime date){
			return Globalization.FormatIsoDate(date);
		}

	}

}